core/src/main/kotlin/com/samples/verifier/internal/SamplesVerifierInstance.kt

```diff
@@ -77,31 +91,48 @@ internal class SamplesVerifierInstance(compilerUrl: String, kotlinEnv: KotlinEnv
       Regex(it.pattern + File.separator + ".*")
     }
     val fileTree = directory.walkTopDown()
-      .onEnter { (configuration.ignoreDirectory == null ||
+      .onEnter {
+        (configuration.ignoreDirectory == null ||
           configuration.ignoreDirectory?.matches(it.relativeTo(directory).toString()) != true)
       }
     for (file in fileTree) {
       val dir = file.relativeTo(directory).toString()
       if (fileRegex?.matches(dir) == false) continue
-      val fileSnippets = when (type) {
-        FileType.MD -> {
-          if (file.extension == "md") {
-            logger.info("Processing ${file}...")
-            processMarkdownFile(file, configuration)
-          } else emptyList()
-        }
-        FileType.HTML -> {
-          if (file.extension == "html") {
-            logger.info("Processing ${file}...")
-            processHTMLFile(file, configuration)
-          } else emptyList()
-        }
-      }
-      snippets.addAll(fileSnippets.withIndex().map { code ->
-        CodeSnippet("${file.nameWithoutExtension}_${code.index}", code.value)
-      })
+      snippets.addAll(processFile(file, type))
+    }
+    return snippets
+  }
 
+  private fun processFiles(directory: File, filenames: List<String>, type: FileType): List<CodeSnippet> {
+    val snippets = mutableListOf<CodeSnippet>()
+    val (fileRegex, ignoreRegex) = listOf(configuration.parseDirectory, configuration.ignoreDirectory).map {
+      it?.let { Regex(it.pattern + File.separator + ".*") }
+    }
+    for (filename in filenames) {
+      if (fileRegex?.matches(filename) == false) continue
```

Comment:
I guess it looks like a `.filter {}` operation with processing the collection (filenames)